decompiler/structures/pseudo/expressions.py

```diff
@@ -427,56 +379,67 @@ class GlobalVariable(Variable):
     """Represents a global variable that comes from MLIL_CONST_PTR.
     MLIL_CONST_PTR represents the following types of pointers:
         - Pointers in .text/.bss/.rodata/.data/symbol table.
-        - Function call, and thereby function pointers."""
+        - Function call, and thereby function pointers.
+    """
 
     def __init__(
         self,
         name: str,
-        vartype: Type = UnknownType(),
+        vartype: Type,
+        initial_value: Expression,  # Technically just UnaryOperation, Constant (with all child classes), and GlobalVariable
         ssa_label: int = None,
         is_aliased: bool = True,
         ssa_name: Optional[Variable] = None,
-        initial_value: Union[float, int, str, GlobalVariable] = None,
+        is_constant: bool = False,
         tags: Optional[Tuple[Tag, ...]] = None,
     ):
         """Init a new global variable. Compared to Variable, it has an additional field initial_value.
         :param initial_value: Can be a number, string or GlobalVariable."""
         super().__init__(name, vartype, ssa_label, is_aliased, ssa_name, tags=tags)
         self.initial_value = initial_value
+        self.is_constant = is_constant
 
     def copy(
         self,
         name: str = None,
         vartype: Type = None,
+        initial_value: Expression = None,
         ssa_label: int = None,
         is_aliased: bool = None,
         ssa_name: Optional[Variable] = None,
-        initial_value: Union[float, int, str, GlobalVariable] = None,
+        is_constant: bool = None,
         tags: Optional[Tuple[Tag, ...]] = None,
     ) -> GlobalVariable:
         """Provide a copy of the current Variable."""
 
         return self.__class__(
             self._name[:] if name is None else name,
             self._type.copy() if vartype is None else vartype,
+            self.initial_value.copy() if initial_value is None else initial_value.copy(),
             self.ssa_label if ssa_label is None else ssa_label,
             self.is_aliased if is_aliased is None else is_aliased,
             self.ssa_name if ssa_name is None else ssa_name,
-            self._get_init_value_copy(initial_value),
+            self.is_constant if is_constant is None else is_constant,
             self.tags if tags is None else tags,
         )
 
+    def __iter__(self) -> Iterator[Expression]:
+        match self.initial_value:
+            case Constant():
+                if isinstance(self.type, ArrayType):
+                    yield from self.initial_value
```

Comment:
Ups legacy code. 
That was used when i tested with representing Arrays as lists of Constants without a new class.
Will be removed.
Should only yield `.initial_value` as it's always an expression

And no the iterator is needed.
Otherwise the `GlobalDeclarationGenerator` visitor does not work properly with initial values.